### PR TITLE
rgw/dbstore object APIs

### PR DESCRIPTION
**Pull request by soumyakoduri**
This PR contains minimal support of Object operations - PUT, GET, DELETE, LIST for regular objects.

### CODE REVIEW

File: src/rgw/store/dbstore/common/dbstore.h [hunk 1]
```diff
@@ -872,6 +1268,364 @@ class DB {
...
+
+          DeleteResult() : delete_marker(false) {}
+        } result;
+
+        explicit Delete(DB::Object *_target) : target(_target) {}
+
+        int delete_obj(const DoutPrefixProvider *dpp);
+      };
+
+      struct Stat {
```
Comment (dang) [inline src/rgw/store/dbstore/common/dbstore.h:1562]: Are these still used?  StatOp is gone from SAL.

**Review by dang — COMMENTED** [review]
This is looking really good.

**Review by dang — APPROVED** [review]
Looks good to me.

File: src/rgw/store/dbstore/common/dbstore.h (hunk 1 above)
Comment (soumyakoduri) [reply src/rgw/store/dbstore/common/dbstore.h]: Missed it.. addressed now.

**Comment by soumyakoduri** [comment]
https://pulpito.ceph.com/soumyakoduri-2021-09-20_19:22:02-rgw-wip-skoduri-dbstore-object-distro-basic-smithi

**Pull request merged**